WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt

```diff
@@ -510,39 +512,46 @@ class StoryComposerActivity : ComposeLoopFrameActivity(),
                 // image yet) so, let's use that for now, and assign the temporaryID we'll use to send
                 // save progress events to Gutenberg.
                 null -> {
-                    val storyMediaFileData =
-                            saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryIdNoMediaFile(
-                                    temporaryId = "$storyIndex-$frameIndex",
-                                    url = if (frame.source is FileBackgroundSource) {
-                                        (frame.source as FileBackgroundSource).file.toString()
-                                    } else {
-                                        (frame.source as UriBackgroundSource).contentUri.toString()
-                                    },
-                                    isVideo = (frame.frameItemType is VIDEO)
-                            )
+                    val storyMediaFileData = buildStoryMediaFileDataForTemporarySlide(
+                            frame,
+                            assignedTempId
+                    )
                     frame.id = storyMediaFileData.id
                     storyMediaFileDataList.add(storyMediaFileData)
                 }
-                // if the frame.id is populated, this should be an actual MediaModel mediaId so,
+                // if the frame.id is populated and is not a temporary id, this should be an actual MediaModel mediaId so,
                 // let's use that to obtain the mediaFile and then replace it with the temporary frame.id
                 else -> {
                     frame.id?.let {
-                        val mediaModel = mediaStore.getSiteMediaWithId(site, it.toLong())
-                        val mediaFile = fluxCUtilsWrapper.mediaFileFromMediaModel(mediaModel)
-                        mediaFile?.let {
-                            val storyMediaFileData =
-                                    saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryId(
-                                            mediaFile = it,
-                                            temporaryId = "$storyIndex-$frameIndex"
-                                    )
-                            frame.id = storyMediaFileData.id
+                        if (it.startsWith(TEMPORARY_ID_PREFIX)) {
+                            val storyMediaFileData = buildStoryMediaFileDataForTemporarySlide(
+                                    frame,
+                                    it
+                            )
+                            // frame.id = storyMediaFileData.id
                             storyMediaFileDataList.add(storyMediaFileData)
+                        } else {
+                            val mediaModel = mediaStore.getSiteMediaWithId(site, it.toLong())
+                            val mediaFile = fluxCUtilsWrapper.mediaFileFromMediaModel(mediaModel)
+                            mediaFile?.let {
```

Comment:
Just a quick question here. If this `mediaFile` is null, what should happen here or is it a case where it's highly unlikely that will happen so we don't have to handle that case in any way?

Reply:
right - as you said, it is highly unlikely for this to happen, given the frame has an id it means it has previously been assigned from a MediaModel, and as such getting a mediaFile from it is trivial and guaranteed to succeed. Given this is only called when the Save button is pressed when editing, the mediaModel exists (and things are recreated from there at this point).

Reply:
Yes, exactly so :) Thank you.